File: javascript/workflow/src/verbs/copy.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import type { CopyArgs } from '@datashaper/schema'

import type { ColumnTableStep } from './util/factories.js'
import { stepVerbFactory } from './util/factories.js'

export const copyStep: ColumnTableStep<CopyArgs> = (input, { column, to }) => {
	return input
		.params({
			column,
		})
		.derive({ [to]: (d: any, $: any) => d[$.column] })
}

export const copy = stepVerbFactory(copyStep)

File: python/datashaper/datashaper/verbs/copy.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
"""Copy verb implementation."""

import pandas as pd
from reactivedataflow import ConfigPort, InputPort, verb

from datashaper import DEFAULT_INPUT_NAME

from .decorators import OutputMode, copy_input_tables, wrap_verb_result


@verb(
    name="copy",
    ports=[
        InputPort(name=DEFAULT_INPUT_NAME, parameter="table", required=True),
        ConfigPort(name="to", required=True),
        ConfigPort(name="column", required=True),
    ],
    adapters=[
        copy_input_tables("table"),
        wrap_verb_result(mode=OutputMode.Table),
    ],
)
def copy(
    table: pd.DataFrame,
    to: str,
    column: str,
) -> pd.DataFrame:
    """Copy verb implementation."""
    table[to] = table[column]
    return table
